std/ascii.d

```diff
@@ -228,14 +347,38 @@ unittest
 
 
 /++
-    Whether or not $(D c) is a punctuation character. That includes all ASCII
-    characters which are not control characters, letters, digits, or whitespace.
+    Params: c = The character to test.
+    Returns: Whether or not $(D c) is a punctuation character. That includes
+    all ASCII characters which are not control characters, letters, digits, or
+    whitespace.
   +/
 bool isPunctuation(dchar c) @safe pure nothrow @nogc
 {
     return c <= '~' && c >= '!' && !isAlphaNum(c);
 }
 
+///
+unittest
+{
+    assert( isPunctuation('.'));
+    assert( isPunctuation(','));
+    assert( isPunctuation(':'));
+    assert( isPunctuation('!'));
+    assert( isPunctuation('#'));
+    assert( isPunctuation('~'));
+    assert( isPunctuation('+'));
+    assert( isPunctuation('_'));
+
+    assert(!isPunctuation('1'));
+    assert(!isPunctuation('a'));
+    assert(!isPunctuation(' '));
+    assert(!isPunctuation('\n'));
+    assert(!isPunctuation('\0'));
```

Comment:
Well, the idea is to include stuff like '~', '+', and '_' that may not be obviously "punctuation", at least not in the traditional sense.